scripts/publish.sh

```diff
@@ -21,13 +21,33 @@ IMAGE_SHA_MATCHED="FALSE"
 AWS_FOR_FLUENT_BIT_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_VERSION)
 AWS_FOR_FLUENT_BIT_STABLE_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_STABLE_VERSION)
 
+PUBLISH_LATEST=$(cat ../linux.version | jq -r '.linux.latest')
+echo "Publish Latest? ${PUBLISH_LATEST}"
+
+# Problem: when we push a new version bump the version number in AWS_FOR_FLUENT_BIT_VERSION file changes
+# but that version is not published immediately. Thus, sync tasks normally
+# sync latest version found in DockerHub. But what if we want to release a non-latest version?
+# then sync tasks need to know this. So we use the script to check for an already published version
+# that's not latest
 docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
 tag_array=(`echo ${docker_hub_image_tags}`)
-AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux ${tag_array[@]})
+AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux latest ${tag_array[@]})
```

Comment:
tag_array[@] only has the last 100 tags.

see
```
curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=1050' | jq -r '.results[].name' | wc -l
```

I think this should be okay for the given logic. I think more recent tags appear at the top, and so this means that if we  publish multiple times we catch the previous publish

Reply:
good point we are relying on behavior in DH API. I will add a comment on this

Reply:
We also only have 64 tags so far..

Reply:
we probably won't reach 100 tags till next year

Reply:
ECS Agent has more than 100 tags, and it does indeed print the latest tags:

```
147dda285fd4:aws-for-fluent-bit wppttt$ curl -s -S  'https://registry.hub.docker.com/v2/repositories/amazon/amazon-ecs-agent/tags/?page=1&page_size=250' | jq -r '.results[].name' | grep 1.69
v1.69.0
arm64-v1.69.0
amd64-v1.69.0
147dda285fd4:aws-for-fluent-bit wppttt$ curl -s -S  'https://registry.hub.docker.com/v2/repositories/amazon/amazon-ecs-agent/tags/?page=1&page_size=250' | jq -r '.results[].name' | grep 1.68
v1.68.2
arm64-v1.68.2
amd64-v1.68.2
v1.68.1
arm64-v1.68.1
amd64-v1.68.1
v1.68.0
arm64-v1.68.0
amd64-v1.68.0
147dda285fd4:aws-for-fluent-bit wppttt$ curl -s -S  'https://registry.hub.docker.com/v2/repositories/amazon/amazon-ecs-agent/tags/?page=1&page_size=250' | jq -r '.results[].name' | grep 1.1
147dda285fd4:aws-for-fluent-bit wppttt$ curl -s -S  'https://registry.hub.docker.com/v2/repositories/amazon/amazon-ecs-agent/tags/?page=1&page_size=250' | jq -r '.results[].name' | grep 1.15
```